data/data-pipeline/data_pipeline/etl/score/etl_score_post.py

```diff
@@ -159,19 +163,25 @@ def _create_score_data(
         states_df: pd.DataFrame,
         score_df: pd.DataFrame,
     ) -> pd.DataFrame:
-        # merge state with counties
-        logger.info("Merging state with county info")
-        county_state_merged = counties_df.merge(
-            states_df, on="State Abbreviation", how="left"
-        )
 
-        # merge state + county with score
-        score_county_state_merged = score_df.merge(
-            county_state_merged,
+        logger.info("Merging county info with score info")
+        score_county_merged = score_df.merge(
+            # We drop state abbreviation so we don't get it twice
+            counties_df[["GEOID", "County Name"]],
             on="GEOID",  # GEOID is the county ID
             how="left",
         )
 
+        logger.info("Merging state info with county-score info")
+        # Here, we need to join on a separate key, since there's no
+        # GEOID for the territories.
+        score_county_merged["State Code"] = score_county_merged[
+            self.GEOID_TRACT_FIELD_NAME
+        ].str[:2]
```

Comment:
We have to create this for the join. If we adopt this method we should probably change the section of code that drops fips codes to use the State Code information rather than `GEOID10_TRACT.str.startswith`